packages/com.influxdata.telegraf/plugins/outputs/OpenTelemetryOutput.pkl

```diff
@@ -0,0 +1,76 @@
+//===----------------------------------------------------------------------===//
+// Copyright © 2024 Apple Inc. and the Pkl project authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//===----------------------------------------------------------------------===//
+/// Sends metrics to [OpenTelemetry](https://opentelemetry.io) servers and agents via gRPC.
+///
+/// <https://github.com/influxdata/telegraf/tree/master/plugins/outputs/opentelemetry>
+@ModuleInfo { minPklVersion = "0.24.0" }
+open module com.influxdata.telegraf.plugins.outputs.OpenTelemetryOutput
+
+extends "Output.pkl"
+
+/// The endpoint to which the metrics will be sent.
+///
+/// Default: `"localhost:4317"`
+service_address: String?
+
+/// The timeout for the connection.
+/// Default: `5.s`
+timeout: Duration?
+
+/// Optional TLS Config.
+/// All variables should specify full path to the file.
+/// Root certificates for verifying server certificates encoded in PEM format.
+tls_ca: String?
+
+/// The public and private key pairs for the client encoded in PEM format.
+/// May contain intermediate certificates.
+tls_cert: String?
+tls_key: String?
+
+/// Send the specified TLS server name via SNI.
+tls_server_name: String?
+
+/// Insecure option to skip TLS server cert verification.
+///
+/// Warning: it is insecure to enable this option.
+/// If enabled, crypto/tls accepts any certificate presented by the server and any host name in that certificate.
+/// Default: `false`
+insecure_skip_verify: Boolean?
+
+/// Send data to a [Coralogix](https://coralogix.com) server.
+coralogix: Coralogix?
+
+/// Set the compression method used to send data.
+///
+/// Default: `"none"`
+compression: ("none"|"gzip")?
```

Comment:
If the default is `"none"` this could just be `compression: *"none"|"gzip"` right?

Reply:
Not quite; annotations in documentation of `Default: x` inform users that the system being configured will default to that value if no value is given. Encoding that default in the Pkl template is dangerous; if the system's default changes, users of the Pkl template will (quietly!) not get that change. In my mind, you should only encode the default if the Pkl template is the source of truth (i.e. maintained by the same folks and in-sync with the system being configured).